### PR TITLE
feat(ThreadMemberManager): Support pagination fetching

### PR DESCRIPTION
**Pull request by Jiralite**
**Please describe the changes this PR makes and why it should be merged:**

- https://github.com/discord/discord-api-docs/pull/5834

**Status and versioning classification:**

- Code changes have been tested against the Discord API, or there are no code changes
- I know how to update typings and have done so, or typings don't need updating
- This PR changes the library's interface (methods or parameters added)

### CODE REVIEW

**Review by Syjalo — CHANGES_REQUESTED** [review]

**Comment by suneettipirneni** [comment]
Perhaps at some point we can make this an async iterator for fetching pages.

File: packages/discord.js/src/managers/ThreadMemberManager.js [hunk 1]
```diff
@@ -130,24 +134,30 @@ class ThreadMemberManager extends CachedManager {
...
-  async _fetchSingle({ member, cache, force = false }) {
+  async _fetchSingle({ member, withMember, cache, force = false }) {
     if (!force) {
       const existing = this.cache.get(member);
       if (existing) return existing;
     }
 
-    const data = await this.client.rest.get(Routes.threadMembers(this.thread.id, member));
+    const data = await this.client.rest.get(Routes.threadMembers(this.thread.id, member), {
+      query: makeURLSearchParams(withMember),
```
Comment (almeidx) [inline packages/discord.js/src/managers/ThreadMemberManager.js:150]:
```suggestion
      query: makeURLSearchParams({ with_member: withMember }),
```

File: packages/discord.js/src/managers/ThreadMemberManager.js [hunk 2]
```diff
@@ -130,24 +134,30 @@ class ThreadMemberManager extends CachedManager {
...
+    });
+
     return this._add(data, cache);
   }
 
   async _fetchMany(options = {}) {
-    const data = await this.client.rest.get(Routes.threadMembers(this.thread.id));
+    const data = await this.client.rest.get(Routes.threadMembers(this.thread.id), {
+      query: makeURLSearchParams(options),
```
Comment (almeidx) [inline packages/discord.js/src/managers/ThreadMemberManager.js:158]:
```suggestion
      query: makeURLSearchParams({ with_member: withMember }),
```

File: packages/discord.js/src/managers/ThreadMemberManager.js [hunk 3]
```diff
@@ -130,24 +134,30 @@ class ThreadMemberManager extends CachedManager {
...
+      query: makeURLSearchParams({ with_member: withMember }),
+    });
+
     return this._add(data, cache);
   }
 
   async _fetchMany(options = {}) {
-    const data = await this.client.rest.get(Routes.threadMembers(this.thread.id));
+    const data = await this.client.rest.get(Routes.threadMembers(this.thread.id), {
+      query: makeURLSearchParams({ with_member: withMember }),
```
Comment (almeidx) [inline packages/discord.js/src/managers/ThreadMemberManager.js:158]: My bad
```suggestion
      query: makeURLSearchParams({ with_member: options.withMember }),
```

Comment (Jiralite) [reply packages/discord.js/src/managers/ThreadMemberManager.js:158]: Ah! Thank you!

File: packages/discord.js/src/structures/ThreadMember.js [hunk 4]
```diff
@@ -35,12 +35,22 @@ class ThreadMember extends Base {
...
-  _patch(data) {
+  _patch(data, extra = {}) {
     if ('join_timestamp' in data) this.joinedTimestamp = Date.parse(data.join_timestamp);
     if ('flags' in data) this.flags = new ThreadMemberFlagsBitField(data.flags).freeze();
+
+    /**
+     * The guild member associated with this thread member.
+     * @type {?GuildMember}
+     */
+    if ('member' in data) {
```
Comment (jaw0r3k) [inline packages/discord.js/src/structures/ThreadMember.js:49]: Are the docs still recognised if they are above `if`?

```suggestion
    if ('member' in data) {
      /**
       * The guild member associated with this thread member.
       * @type {?GuildMember}
       */
```

**Review by kyranet — APPROVED** [review]

File: packages/discord.js/src/structures/ThreadMember.js [hunk 5]
```diff
@@ -35,12 +35,23 @@ class ThreadMember extends Base {
...
   }
 
-  _patch(data) {
+  _patch(data, extra = {}) {
     if ('join_timestamp' in data) this.joinedTimestamp = Date.parse(data.join_timestamp);
     if ('flags' in data) this.flags = new ThreadMemberFlagsBitField(data.flags).freeze();
+
+    if ('member' in data) {
+      /**
+       * The guild member associated with this thread member.
```
Comment (vladfrangu) [inline packages/discord.js/src/structures/ThreadMember.js:47]: Well this won't be confusing with the other aptly named `guildMember` getter 😢

Comment (Jiralite) [reply packages/discord.js/src/structures/ThreadMember.js:47]: This one is `@private`. The getter first resolves this property before attempting to retrieve from the cache.

**Review by iCrawl — APPROVED** [review]

**Review by SpaceEEC — APPROVED** [review]

**Pull request merged**